specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2023-04-01-preview/agfood.json

```diff
@@ -18095,6 +18044,27 @@
     }
   },
   "definitions": {
+    "ApiKeyAuthCredentials": {
+      "description": "ApiKeyAuthCredentials.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AuthCredentials"
+        },
+        {
+          "required": [
+            "apiKey"
+          ],
+          "type": "object",
+          "properties": {
+            "apiKey": {
+              "$ref": "#/definitions/KeyVaultProperties"
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "ApiKeyAuth"
```

Comment:
The [Azure Naming Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#common-names) recommend that the discriminator field be named `kind`